typed-protocols/src/Network/TypedProtocol/Pipelined.hs

```diff
@@ -16,6 +16,7 @@ module Network.TypedProtocol.Pipelined
   , Nat (Zero, Succ)
   , natToInt
   , unsafeIntToNat
+  , fmapPeerPipelined
```

Comment:
If we're adding something like this, we should do it consistently for the non-pipelined case too.

We want to publish this package too.

Reply:
Sure, so you'd like to export an `id` function? ;)
```
fmapPeer :: (Peer ps  pr st  m a -> Peer ps' pr st' m b)
         -> Peer ps  pr st  m a
         -> Peer ps' pr st' m b
fmapPeer f = f
```

Reply:
But what we can do is to add  'Functor' instances to both 'Peer' and 'PeerPipelined'.

Reply:
Fair point, that wouldn't make sense. If we can do `Functor` instances though then yes we should do that.

Reply:
An `Applicative` instance be useful too. It would give poor man's multiplexer in `IO`, but a much nicer one using `Concurrently` wrapper from `async`.   But that's for another PR.

Reply:
Hmm, writing `pure :: a -> m a` requires to know which state is the final one, some protocols might not even have a single final state.

Reply:
Also `ap` is only possible for some protocols.  it's type signature is similar to `connect`.